py/objnamedtuple.c

```diff
@@ -52,6 +52,23 @@ STATIC size_t namedtuple_find_field(const mp_obj_namedtuple_type_t *type, qstr n
     return (size_t)-1;
 }
 
+#if MICROPY_PY_COLLECTIONS_ORDEREDDICT && MICROPY_CPYTHON_COMPAT
```

Comment:
This should be something like MICROPY_PY_COLLECTIONS_NAMEDTUPLE__ASDICT, disabled by default, enabled in coverage build, properly running in tests for both builds.